src/scene/lightmap.js

```diff
@@ -0,0 +1,277 @@
+pc.extend(pc, function () {
+
+    var maxSize = 2048;
+    var maskDynamic = 1;
+    var maskBaked = 2;
+    var maskLightmap = 4;
+
+    var sceneLightmaps = [];
+    var lmCamera;
+
+    function collectModels(node, nodes) {
+        if (node.model) {
+            if (node.model.data.lightmapped) {
+                nodes.push(node);
+            }
+        }
+        var children = node.getChildren();
+        for(var i=0; i<children.length; i++) {
+            collectModels(children[i], nodes);
+        }
+    }
+
+    var Lightmapper = function (device, root, scene, renderer) {
```

Comment:
I think this should be LightMapper instead of Lightmapper. I believe the rest of the engine follows this convention where possible.